File: TransCoder_tokenized_test_set_functions/java/QUERIES_ON_ARRAY_WITH_DISAPPEARING_AND_REAPPEARING_ELEMENTS.java
static void PerformQueries ( int [ ] a, int [ ] [ ] vec ) {
  Vector < Integer > ans = new Vector < > ( );
  int n = ( int ) a . length - 1;
  int q = ( int ) vec . length;
  for ( int i = 0;
  i < q;
  ++ i ) {
    long t = vec [ i ] [ 0 ];
    int m = vec [ i ] [ 1 ];
    if ( m > n ) {
      ans . add ( - 1 );
      continue;
    }
    int turn = ( int ) ( t / n );
    int rem = ( int ) ( t % n );
    if ( rem == 0 && turn % 2 == 1 ) {
      ans . add ( - 1 );
      continue;
    }
    if ( rem == 0 && turn % 2 == 0 ) {
      ans . add ( a [ m ] );
      continue;
    }
    if ( turn % 2 == 0 ) {
      int cursize = n - rem;
      if ( cursize < m ) {
        ans . add ( - 1 );
        continue;
      }
      ans . add ( a [ m + rem ] );
    }
    else {
      int cursize = rem;
      if ( cursize < m ) {
        ans . add ( - 1 );
        continue;
      }
      ans . add ( a [ m ] );
    }
  }
  for ( int i : ans ) System . out . print ( i + "\n" );
}

File: TransCoder_tokenized_test_set_functions/python/QUERIES_ON_ARRAY_WITH_DISAPPEARING_AND_REAPPEARING_ELEMENTS.py
def PerformQueries ( a, vec ) :
    ans = [ ];
    
    n = len ( a ) - 1;
    
    q = len ( vec );
    
    for i in range ( q ) :
        t = vec [ i ] [ 0 ];
        
        m = vec [ i ] [ 1 ];
        
        if ( m > n ) :
            ans . append ( - 1 );
            
            continue;
            
        turn = t //n;
        rem=t%n;
        if(rem==0andturn%2==1):
            ans.append(-1);
            continue;
        if(rem==0andturn%2==0):
            ans.append(a[m]);
            continue;
        if(turn%2==0):
            cursize=n-rem;
            if(cursize<m):
                ans.append(-1);
                continue;
            ans.append(a[m+rem]);
        else:
            cursize=rem;
            if(cursize<m):
                ans.append(-1);
                continue;
            ans.append(a[m]);
    foriinans:
        print(i);
    
